common/rakelib/utils.rake

```diff
@@ -121,19 +122,26 @@ class Utils
   private_class_method :plain
 
   # select the xcode version we want/support
-  def self.version_select
-    version = '8.*'
-    xcodes = `mdfind "kMDItemCFBundleIdentifier = 'com.apple.dt.Xcode' && kMDItemVersion = '#{version}'"`.chomp.split("\n")
-    if xcodes.empty?
-      raise "\n[!!!] SwiftGen requires Xcode #{version}, but we were not able to find it. If it's already installed update your Spotlight index with 'mdimport /Applications/Xcode*'\n\n"
+  @@version_select ||= begin
+    # if current Xcode already fulfills min version don't force DEVELOPER_DIR=…
+    current_xcode = Pathname.new(`xcode-select -p`).parent.parent
+    current_xcode_version = `mdls -name kMDItemVersion -raw #{current_xcode}`.chomp
+    if current_xcode_version.to_f >= MIN_XCODE_VERSION
```

Comment:
Prefer the `return "" if …` construct, similar to a `guard` in Swift, to avoid having to indent the while rest in a big `else`.